charts/astronomer/templates/_helpers.yaml

```diff
@@ -376,20 +376,21 @@ Return  the proper Storage Class
 {{/*
 Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
 but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
+This version prioritizes component-specific registry.persistence.storageClassName over global.storageClass.
 */}}
-{{- if .Values.global.storageClass -}}
-    {{- if (eq "-" .Values.global.storageClass) -}}
+{{- if .Values.registry.persistence.storageClassName -}}
+    {{- if (eq "~" .Values.registry.persistence.storageClassName) -}}
         {{- printf "storageClassName: \"\"" -}}
     {{- else }}
-        {{- printf "storageClassName: %s" .Values.global.storageClass -}}
+        {{- printf "storageClassName: %s" .Values.registry.persistence.storageClassName -}}
     {{- end -}}
 {{- else -}}
-    {{- if .Values.registry.persistence.storageClassName -}}
-          {{- if (eq "~" .Values.registry.persistence.storageClassName) -}}
-              {{- printf "storageClassName: \"\"" -}}
-          {{- else }}
-              {{- printf "storageClassName: %s" .Values.registry.persistence.storageClassName -}}
-          {{- end -}}
+    {{- if .Values.global.storageClass -}}
+        {{- if (eq "-" .Values.global.storageClass) -}}
+            {{- printf "storageClassName: \"\"" -}}
+        {{- else }}
+            {{- printf "storageClassName: %s" .Values.global.storageClass -}}
+        {{- end -}}
```

Comment:
This logic can be drastically simplified using `or`

```
$ cat templates/serviceaccount.yaml
{{- if .Values.serviceAccount.create -}}
apiVersion: v1
kind: ServiceAccount
metadata:
  name: {{ or .Values.foo .Values.bar }}
{{- end }}
$ helm template . | grep name:
  name:
$ helm template . --set foo=first --set bar=second | grep name:
  name: first
$ helm template . --set bar=second | grep name:
  name: second
```

Reply:
done